File: C++/ejercicio5.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

class Evaluacion {
public:
    string nombre;
    double n1, n2, n3, promedio;
    string estado;

    Evaluacion(string nom, double nota1, double nota2, double nota3) {
        nombre = nom; n1 = nota1; n2 = nota2; n3 = nota3;
        promedio = (n1 + n2 + n3) / 3.0;
        estado = (promedio >= 7.0) ? "Aprobado" : "Reprobado";
    }

    string toString() {
        return nombre + "," + to_string(promedio) + "," + estado;
    }
};

int main() {
    string nom; double n1, n2, n3;
    cout << "--- SISTEMA DE CALIFICACIONES ---\n";
    cout << "Nombre del estudiante: "; getline(cin, nom);
    cout << "Nota 1: "; cin >> n1;
    cout << "Nota 2: "; cin >> n2;
    cout << "Nota 3: "; cin >> n3;

    Evaluacion eval(nom, n1, n2, n3);

    ofstream file("calificaciones.txt", ios::app);
    if (file.is_open()) {
        file << eval.toString() << endl;
        file.close();
        cout << "Notas registradas. Promedio: " << eval.promedio << " | Estado: " << eval.estado << "\n";
    }
    return 0;
}

File: Java/ejercicio5.java
import java.io.*;
import java.util.Scanner;

class Evaluacion {
    private String nombre;
    private double promedio;
    private String estado;

    public Evaluacion(String nombre, double n1, double n2, double n3) {
        this.nombre = nombre;
        this.promedio = (n1 + n2 + n3) / 3.0;
        this.estado = (this.promedio >= 7.0) ? "Aprobado" : "Reprobado";
    }

    @Override
    public String toString() {
        return nombre + "," + promedio + "," + estado;
    }
    public double getPromedio() { return promedio; }
    public String getEstado() { return estado; }
}

public class ejercicio5 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("--- REGISTRO DE NOTAS ---");
        System.out.print("Estudiante: "); String nom = sc.nextLine();
        System.out.print("Nota 1: "); double n1 = sc.nextDouble();
        System.out.print("Nota 2: "); double n2 = sc.nextDouble();
        System.out.print("Nota 3: "); double n3 = sc.nextDouble();

        Evaluacion e = new Evaluacion(nom, n1, n2, n3);

        try (PrintWriter pw = new PrintWriter(new FileWriter("calificaciones.txt", true))) {
            pw.println(e.toString());
            System.out.printf("Registro exitoso. Promedio: %.2f [%s]\n", e.getPromedio(), e.getEstado());
        } catch (IOException ex) {
            System.out.println("Error al guardar.");
        }
    }
}
